x-pack/plugins/cases/server/connectors/cases/cases_connector_executor.ts

```diff
@@ -752,28 +746,44 @@ export class CasesConnectorExecutor {
     };
   }
 
-  private getCasesTitle(ruleName: string, oracleCounter: number) {
-    const suffix =
-      oracleCounter === INITIAL_ORACLE_RECORD_COUNTER
-        ? '(Auto-created)'
-        : `(${oracleCounter}) (Auto-created)`;
+  private getCasesTitle(
+    params: CasesConnectorRunParams,
+    grouping: GroupedAlerts['grouping'],
+    oracleCounter: number
+  ) {
+    const groupingDescription = Object.entries(grouping)
+      .map(([_, value]) => {
+        return convertValueToString(value);
+      })
+      .join(' & ');
 
-    const ruleNameTrimmed = ruleName.slice(0, MAX_TITLE_LENGTH - suffix.length - 1);
+    const suffix = `${
+      groupingDescription.length > 0 ? ` - Grouping by ${groupingDescription}` : ''
+    }${oracleCounter > INITIAL_ORACLE_RECORD_COUNTER ? ` (${oracleCounter})` : ''} (Auto-created)`;
 
-    return `${ruleNameTrimmed} ${suffix}`;
+    const ruleNameTrimmed = params.rule.name.slice(0, MAX_TITLE_LENGTH - suffix.length - 1);
+
+    return `${ruleNameTrimmed}${suffix}`;
   }
 
-  private getGroupingDescription(grouping: GroupedAlerts['grouping']) {
-    const flattenGrouping = getFlattenedObject(grouping);
+  private getCaseDescription(params: CasesConnectorRunParams, grouping: GroupedAlerts['grouping']) {
+    const ruleName = params.rule.ruleUrl
+      ? `[${params.rule.name}](${params.rule.ruleUrl})`
+      : params.rule.name;
 
-    return Object.entries(flattenGrouping)
-      .map(([key, value]) => {
-        const keyAsCodeBlock = `\`${key}\``;
-        const valueAsCodeBlock = `\`${convertValueToString(value)}\``;
+    const description = `This case was created by the Case action in ${ruleName}.`;
 
-        return `${keyAsCodeBlock} equals ${valueAsCodeBlock}`;
+    const groupingDescription = Object.entries(grouping)
+      .map(([key, value]) => {
+        return `\`${key}: ${convertValueToString(value)}\``;
       })
       .join(' and ');
+
+    if (groupingDescription.length > 0) {
+      return `${description} The assigned alerts are grouped by ${groupingDescription}.`;
+    }
+
+    return `${description}`;
```

Comment:
shouldn't this be just `return description`?

Reply:
Done [`af0baec` (#180758)](https://github.com/elastic/kibana/pull/180758/commits/af0baecbd806b3643dd3265d37c0cf5e30916ebf)